en/docs/learn/api-controller/advanced-topics/configuring-environment-specific-parameters.md

```diff
@@ -0,0 +1,82 @@
+
+The following is the structure of the parameter file.
+
+```go
+environments:
+    - name: <environment_name>
+      endpoints:
+          production:
+              url: <production_endpoint_url>
+              config:
+                  retryTimeOut: <no_of_retries_before_suspension>
+                  retryDelay: <retry_delay_in_ms>
+                  factor: <suspension_factor>
+          sandbox:
+              url: <sandbox_endpoint_url>
+              config:
+                  retryTimeOut: <no_of_retries_before_suspension>
+                  retryDelay: <retry_delay_in_ms>
+                  factor: <suspension_factor>
+      security:
+        - enabled: <whether_security_is_enabled>
+          type: <endpoint_authentication_type_basic_or_digest>
+          username: <endpoint_username>
+          password: <endpoint_password>
+      gatewayEnvironments:
+        - <gateway_environment_name>           
+      certs:
+        - hostName: <endpoint_url>
+          alias: <certificate_alias>
+          path: <certificate_file_path>
+```
+The following code snippet contains sample configuration of the parameter file.
+
+!!! example
+    ```go
+    environments:
+        - name: dev
+          endpoints:
+            production:
+              url: 'https://dev.wso2.com'
+          security:
+           - enabled: true
+             type: basic
+             username: admin
+             password: admin
+          certs:
+            - hostName: 'https://dev.wso2.com'
+              alias: Dev
+              path: ~/.certs/dev.pem 
+          gatewayEnvironments:
+            - Production and Sandbox    
+        - name: test
+          endpoints:
+            production:
+              url: 'https://test.wso2.com'
+          security:
+           - enabled: true
+             type: digest
+             username: admin
+             password: admin
+              config:
+                retryTimeOut: $RETRY
+            sandbox:
+              url: 'https://test.sandbox.wso2.com'
+    ```
+Instead of the default `api_params.yaml`, you can a provide custom parameter file using `--params` flag. A sample command will be as follows.
+
+!!! example
+    ```go
+    apictl import-api -f dev/PhoneVerification_1.0.zip -e production --params /home/user/custom_params.yaml 
+    ```
+
+!!! info
+    -   Production/Sandbox backends for each environment can be specified in the parameter file with additional configurations, such as timeouts.
+    -   Under the security field, if the `enabled` attribute is `true`, you must specify the `username`, `password` and the `type` (can be either only `basic` or `digest`). If the `enabled` attribute is `false`, then non of the security parameters will be set. If the `enabled` attribute is not set (blank), then the security parameters in api.yaml file will be considered.
```

Comment:
In the middle of the second line, shall we change "non" to "none"?

Reply:
This is already fixed in the master branch via https://github.com/wso2/docs-apim/commit/c049f0216cc3f99dbd8b9cfd5d3021c318f365d2
Hence we can ignore this.

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3098/commits/edd8ae4c020155b3c9d8e4465a923a031e8f9472 for 3.2.0

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3099/commits/5f5c672fdf5973c1368873c40f65a138e2fbe58a for 3.1.0